packages/db/src/loaders/run.ts

```diff
@@ -2,26 +2,25 @@ import { logger } from "@truffle/db/logger";
 const debug = logger("db:loaders:run");
 
 import { promisify } from "util";
-import type Web3 from "web3";
+import type {Provider} from "web3/providers";
 import {DocumentNode} from "graphql";
 
 import { Loader, LoadRequest, GraphQlRequest, Web3Request, RequestType } from "./types";
 
-interface ITruffleDB {
-  query: (query: DocumentNode | string, variables: any) => Promise<any>;
-}
-
-export type LoaderRunner = <
-  A extends unknown[],
+export type LoaderRunner = < A extends unknown[],
   T = any,
   R extends RequestType | undefined = undefined
 >(
   loader: Loader<A, T, R>,
   ...args: A
 ) => Promise<T>;
 
-export const forDb = (db): {
-  forProvider(provider: Web3["currentProvider"]): {
+export interface Db {
+  query: (query: DocumentNode | string, variables: any) => Promise<any>;
```

Comment:
Do these have to be `any`? I guess we don't know what they'll be ahead of time per se, but could we use `unknown` instead?

Reply:
Hm, we _could_ know it ahead of time, but I put it as `any` out of laziness

Reply:
I don't feel super strongly about it but I have been trying to avoid using `any` where I can.

Reply:
Definitely. `unknown` isn't much better, unfortunately. I'm inclined to let this slide, though, because this is kind of just passed to GraphQL sight-unseen, so it's whatever they say is allowable.

Also, GraphQL validates all requests coming in at runtime, so I'm less concerned about type-safety at this boundary point.